src/app/core/models/notification-options.model.ts

```diff
@@ -0,0 +1,11 @@
+import { TemplateRef } from '@angular/core';
+
+export class NotificationOptions {
+  type: Notification['type'];
```

Comment:
`Notification['type']` looks like some weird approach... TS has a lot of another ways to use types more elegant.